packages/esm-patient-forms-app/src/forms/forms-see-all.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import dayjs from 'dayjs';
+import first from 'lodash-es/first';
+import ContentSwitcher from 'carbon-components-react/es/components/ContentSwitcher';
+import Switch from 'carbon-components-react/es/components/Switch';
+import FormView from './form-view-see-all.component';
+import styles from './forms.component.scss';
+import { EmptyState, ErrorState } from '@openmrs/esm-patient-common-lib';
+import { useTranslation } from 'react-i18next';
+import { fetchAllForms, fetchPatientEncounters } from './forms.resource';
+import { filterAvailableAndCompletedForms } from './forms-utils';
+import { Encounter, Form } from '../types';
+import EmptyFormView from './empty-form.component';
+import { getObjectFHIR } from './get-FHIR-object';
+
+enum FormViewState {
+  recommended = 0,
+  completed,
+  all,
+}
+
+const Forms = () => {
+  const urlPathArray = window.location.pathname.split('/');
+  const pathUuid = urlPathArray[urlPathArray.length - 2];
+  const [patientUuid, setUUID] = React.useState(null);
+  const [patient, setPatient] = React.useState<fhir.Patient>();
+
+  React.useEffect(() => {
+    getObjectFHIR(patientUuid).then(setPatient);
+    setUUID(pathUuid);
```

Comment:
Why is this needed? I think the UUID is already known and does not change. Also let's not use `window.location` to read that out... `useHistory` and get the matching param instead.

Reply:
Is this `useHistory` from "react-router-dom" ? and does this involve wrapping the page with the <Router>  component? could you please point me to a script that has used `useHistory` inside the 3.0 framework?

Reply:
It is from `react-router-dom`. There should be plenty of examples using `useHistory` / `react-router-dom` in this repository. Please look for it yourself (hint: there should be a search functionality in whatever IDE you are using).

Reply:
I have implemented the other changes but in spite of a lot of attempts “useHistory” was giving errors as reported in “https://stackoverflow.com/questions/64838587/how-to-properly-use-usehistory-from-react-router-dom” so could not make use of it. A proper solution to the error could not be found anywhere, so continued with the old implementation.